soccer/gameplay/tactics/wall.py

```diff
@@ -76,84 +79,93 @@ def _add_wall_defenders(self):
         # prioritize the middle elements of the wall
         for i in range(len(pts)):
             pt = pts[i]
+            subbhvr_name = f"robot{i}"
+            if self.has_subbehavior_with_name(subbhvr_name):
+                self.remove_subbehavior(subbhvr_name)
             self.add_subbehavior(self.WallMove(
                 pt, pt if i != 0 and i != len(pts) - 1 else None),
-                                 name="robot" + str(i),
+                                 name=subbhvr_name,
                                  required=False)
 
     ## Remove wall behaviors
-    def _remove_wall_defenders(self):
+    def _remove_wall_defenders(self) -> None:
         self.update_midpoint()
-        for i in range(len(self.active_defenders)):
+        for i in range(self.active_defenders):
             name = "robot" + str(i)
             pt = self.calculate_destination(i)
             if i < self.number_of_defenders:
-                self.subbehavior_with_name(name).pos = pt
+                move_subbhvr = self.subbehavior_with_name(name)
+                assert isinstance(move_subbhvr, self.WallMove)
```

Comment:
I'll comment here to bring up a conversation about asserts in our python code. @kylestach @chachmu @liquidmetal9015 @HussainGynai 

When things break, it's more obvious where, but if something fucks up in comp and it may have still worked, it'll cause a blacklist loop and we'll have to timeout

Reply:
What about a util that just changes between real assert and no op depending on whether we're in an actual game or not?

Reply:
I usually tend towards assert for states that should be unreachable (aka statically-guaranteed). This should be one of those cases - basically, if this is failing from a particular call site, it will _always_ fail from that call site, so it is okay.